### ISSUE TITLE
onerror for Elements is incorrect

### ISSUE THREAD

**Issue by trusktr** [microsoft/TypeScript-DOM-lib-generator#1821]
```ts
const img = document.createElement('img')

// type error:
img.onerror = (e: Event) => e
```

<img width="614" alt="Screenshot 2024-10-07 at 8 03 39 PM" src="https://github.com/user-attachments/assets/ae7ea129-5b59-411d-bbec-49ad054b5dc4">

[playground](https://www.typescriptlang.org/play/?#code/MYewdgzgLgBAlgWwOYwLwwCYmAVwQUzCgDpgAnfAQynwFEAbfAogCgHJEk2BKAKF4D0AmAAkcACwD8k3p2Lh8ZMiDJoYLfAC4YtAG6Eo3NAD4Y+XkA)

Only `Window.onerror` should have the alternative signature, but not elements. See:

- https://developer.mozilla.org/en-US/docs/Web/API/Window/error_event
- https://udn.realityripple.com/docs/Web/API/GlobalEventHandlers/onerror

**Comment by trusktr** [comment]
The reason I'm posting this is because when trying to create JSX types for Custom Elements, it is leaking into the JSX types, causing unexpected errors:

<img width="860" alt="Screenshot 2024-10-07 at 8 06 21 PM" src="https://github.com/user-attachments/assets/ed9f3783-eff2-43fe-8cff-e7f490923b39">

**Comment by saschanaz** [comment]
This is a bit awkward situation as img inherits from GlobalEventHandlers. I wonder we can somehow override it?

**Comment by trusktr** [comment]
@saschanaz I can this code:

```js
const img = document.createElement('img')
img.onerror = (...args) => console.log(args.length, ...args)
img.src = '@#$%'
```

and it logs an `1, Event` (not `1, ErrorEvent`) in Chrome, Edge, and Firefox, and Safari.

Hmm, maybe the parameter type needs to be just `Event`, but it still doesn't need the signature of `GlobalEventHandlers.onerror`.
